src/xcode/ENA/ENA/Source/Services/Risk/Provider/Helper/__tests__/CachedAppConfigurationTests.swift

```diff
@@ -84,6 +84,21 @@ final class CachedAppConfigurationTests: XCTestCase {
 
 		subscription.cancel()
 	}
+	
+	func testClientMetadata_isUpdated_everytime_appconfiguration_isFetched() {
+		let store = MockTestStore()
+		XCTAssertNil(store.clientMetadata, "Client metadata should be initially nil")
+		let client = CachingHTTPClientMock(store: store)
+		let cache = CachedAppConfiguration(client: client, store: store)
+		let expectationClientMetadata = expectation(description: "ClientMetadata")
+		let configuration = cache.appConfiguration(forceFetch: true).sink { _ in
+			expectationClientMetadata.fulfill()
+		}
+		waitForExpectations(timeout: 1) { _ in
+			XCTAssertNotNil(configuration, "configuration is not nil")
+			XCTAssertNotNil(store.clientMetadata, "Client metadata should be filled after fetching")
```

Comment:
You should test here also the values you are expecting, not only if it is nil or not.